File: frontend/start-dev.js
#!/usr/bin/env node
/**
 * Development server startup script for RAG Research Assistant Frontend
 */

const { spawn } = require('child_process');
const path = require('path');
const fs = require('fs');

console.log('🚀 Starting RAG Research Assistant Frontend (Development Mode)');
console.log('📁 Working directory:', process.cwd());
console.log('🔄 Hot reloading enabled');
console.log('🌐 Frontend will be available at: http://localhost:3000');
console.log();

// Check if .env.local file exists
const envFile = path.join(process.cwd(), '.env.local');
if (!fs.existsSync(envFile)) {
    console.log('⚠️  No .env.local file found. Copy .env.example to .env.local and configure:');
    console.log('   cp .env.example .env.local');
    console.log();
}

// Check if node_modules exists
const nodeModules = path.join(process.cwd(), 'node_modules');
if (!fs.existsSync(nodeModules)) {
    console.log('📦 Installing dependencies...');
    const install = spawn('npm', ['install'], { stdio: 'inherit', shell: true });
    
    install.on('close', (code) => {
        if (code === 0) {
            startDevServer();
        } else {
            console.error('❌ Failed to install dependencies');
            process.exit(1);
        }
    });
} else {
    startDevServer();
}

function startDevServer() {
    console.log('🎯 Starting Next.js development server...');
    
    const nextDev = spawn('npm', ['run', 'dev'], { 
        stdio: 'inherit', 
        shell: true,
        env: {
            ...process.env,
            FORCE_COLOR: '1'
        }
    });

    nextDev.on('close', (code) => {
        if (code !== 0) {
            console.error(`❌ Development server exited with code ${code}`);
            process.exit(code);
        }
    });

    process.on('SIGINT', () => {
        console.log('\n👋 Frontend server stopped');
        nextDev.kill('SIGINT');
        process.exit(0);
    });
}

File: backend/start-dev.py
#!/usr/bin/env python3
"""
Development server startup script for RAG Research Assistant Backend
"""

import os
import sys
import subprocess
from pathlib import Path

def main():
    # Set environment to development
    os.environ["ENV"] = "development"
    
    print("🚀 Starting RAG Research Assistant Backend (Development Mode)")
    print("📁 Working directory:", os.getcwd())
    print("🔄 Auto-reload enabled")
    print("🌐 Backend will be available at: http://localhost:8000")
    print("📖 API Documentation: http://localhost:8000/docs")
    print()
    
    # Check if .env file exists
    env_file = Path(".env")
    if not env_file.exists():
        print("⚠️  No .env file found. Copy .env.example to .env and configure:")
        print("   cp .env.example .env")
        print()
    
    try:
        # Run uvicorn with development settings
        subprocess.run([
            sys.executable, "-m", "uvicorn",
            "app:app",
            "--host", "0.0.0.0",
            "--port", "8000",
            "--reload",
            "--reload-exclude", "uploads/*",
            "--reload-exclude", "chroma_data/*",
            "--access-log"
        ], check=True)
    except KeyboardInterrupt:
        print("\n👋 Backend server stopped")
    except subprocess.CalledProcessError as e:
        print(f"\n❌ Error starting server: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()